src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestServiceInstaller.cs

```diff
@@ -130,24 +128,30 @@ public unsafe void Install()
 
         public void RemoveService()
         {
-            if (ServiceName == null)
-                throw new InvalidOperationException($"Already removed service at stack ${_removalStack}");
-
-            // Store the stack for logging in case we're called twice
             try
             {
-                throw new Exception();
+                StopService();
             }
-            catch (Exception e)
+            finally
             {
-                _removalStack = e.StackTrace;
+                // If the service didn't stop promptly, we will get a TimeoutException.
+                // This means the test service has gotten "jammed".
```

Comment:
Is jammed an official term?

Reply:
No I just mean the service did not notify that it has stopped, for whatever reason.